imports.go

```diff
@@ -61,6 +61,39 @@ func (i *Imports) RegisterImport(importPath, packageName string) string {
 	return i.prefixForPackage(importPath, packageName, true)
 }
 
+// RegisterAliasedImport "imports" the specified package with a given alias and
+// returns the package prefix to use for symbols in the imported package. It is
+// safe to import the same package repeatedly using the same alias.
+//
+// If no
+//
+// However, this function will return an error if the given importPath has
+// already been registered and cannot be identified using the provided package
+// name.
+func (i *Imports) RegisterAliasedImport(importPath, packageName string) (string, error) {
```

Comment:
I think the convention for other functions in this package is to panic if used erroneously.

Reply:
The reason for the error is that it's hard for the caller to know ahead of time that the given package name or import path is already registered. Most other calls are OK if a package is added with a different alias.

That said, perhaps the API can change (as mentioned below) instead of adding this method.

Reply:
Outdated, closing out.